test/solvers/smooth/smooth_test.py

```diff
@@ -0,0 +1,133 @@
+# Copyright 2014-2016 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Test for the Chambolle-Pock solver."""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+
+# External
+import pytest
+
+# Internal
+import odl
+from odl.operator import OpNotImplementedError
+from odl.util.testutils import noise_element
+
+
+@pytest.fixture(scope="module", params=['l2_squared', 'l2_squared_scaled'])
+def functional(request):
+    """Return a functional whose optimal value is at 0."""
+    name = request.param
+
+    # TODO: Add rosenbrock (#600) and quadratic (#606) functionals
```

Comment:
Just did this which resulted in a major "review" of the quasi newton solvers. They now have a primitive resetting functionality since they refused to converge due to instabilities due to scale differences.